src/main/java/org/openrewrite/maven/ResourceParser.java

```diff
@@ -106,4 +107,56 @@ public <S extends SourceFile> List<S> parseSourceFiles(
             throw new UncheckedIOException(e);
         }
     }
+
+    public <S extends SourceFile> List<S> parseQuarks(
+            Path baseDir,
+            Path searchDir,
+            Collection<Path> alreadyParsed,
+            ExecutionContext ctx) {
+        QuarkParser parser = new QuarkParser();
+        try (Stream<Path> resources = Files.find(searchDir, 16, (path, attrs) -> {
+            if (path.toString().endsWith(".java")) {
+                return false;
+            }
```

Comment:
Why is this necessary?

Reply:
Oh I see we run `ResourceParser` before `JavaParser`. Makes sense

Reply:
I ran the maven plugin on a few projects and java files are added without the condition.

- `.java` files aren't added to already parsed. I tested using `listJavaSources` to add the paths as already parsed, but the files were still added as quarks.
- The other resource parsers only accept specific types of files `.yaml`, `.java`, `.xml`, but the `QuarkParser#accept()` method always returns `true`.

Reply:
I think a better, long-term solution, would just be to map the paths for all the Java source files into the `alreadyParsed` and not have any specialized logic. I envision a world where there will also be groovy and Kotlin parsers in the mix and I would not want to have separate logic to exclude certain file masks (like "*.java")

Reply:
We do parse the Java sources before the resources (I believe)

Reply:
Is there something specialized in `parseQuarks` that is not handled by `parseSourceFiles`? It seems largely a duplicate of that method.